src/globus_sdk/_remarshal.py

```diff
@@ -0,0 +1,159 @@
+"""
+This module provides internal helpers for remarshalling data from one shape
+into another.
+
+This may be as simple as converting strings from one form to another, or as
+sophisticated as building a specific internal object in a configurable way.
+"""
+
+from __future__ import annotations
+
+import collections.abc
+import typing as t
+import uuid
+
+from globus_sdk._missing import MISSING, MissingType
+from globus_sdk._types import UUIDLike
+
+T = t.TypeVar("T")
+R = t.TypeVar("R")
+
+
+def safe_strseq_iter(
+    value: t.Iterable[t.Any] | str | uuid.UUID,
+) -> t.Iterator[str]:
+    """
+    Given an Iterable (typically of strings), produce an iterator over it of strings.
+
+    :param value: The stringifiable object or objects to iterate over
+
+    This is a passthrough with some caveats:
+    - if the value is a solitary string, yield only that value
+    - if the value is a solitary UUID, yield only that value (as a string)
+    - str values in the iterable which are not strings
+
+    This helps handle cases where a string is passed to a function expecting an iterable
+    of strings, as well as cases where an iterable of UUID objects is accepted for a
+    list of IDs, or something similar.
+    """
+    if isinstance(value, str):
+        yield value
+    elif isinstance(value, uuid.UUID):
+        yield str(value)
+    else:
+        for x in value:
+            yield str(x)
+
+
+def safe_stringify(value: object | MissingType | None) -> str | MissingType | None:
```

Comment:
Out of curiosity, why `object` and not `t.Any` here?

Reply:
`Any` and `object` are really similar in these cases, but I'm trying (personally) to break the habit of using `Any` where only `object`.

In this case it doesn't have an impact. Here's one where it does:
```python
def frob(obj: object) -> str:
    return obj.frobulate()  # error! `object` doesn't provide `.frobulate()`!

def frob(obj: Any) -> str:
    return obj.frobulate()  # pass! you said 'Any', so how can we know if it has `.frobulate()`?
```